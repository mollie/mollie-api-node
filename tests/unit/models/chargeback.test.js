import Chargeback from 'models/chargeback';

describe('chargeback model', () => {
  it('should instantiate with default values', () => {
    const chargeback = new Chargeback();

    expect(chargeback.toPlainObject()).toMatchSnapshot();
  });

  it('should instantiate with given values', () => {
    const chargebackProps = JSON.parse(`{
        "resource": "chargeback",
        "id": "chb_n9z0tp",
        "amount": {
            "currency": "USD",
            "value": "43.38"
        },
        "settlementAmount": {
            "currency": "EUR",
            "value": "-35.07"
        },
        "createdAt": "2018-03-14T17:00:52.0Z",
        "reversedAt": null,
        "paymentId": "tr_WDqYK6vllg",
        "_links": {
            "self": {
                "href": "https://api.mollie.com/v2/payments/tr_WDqYK6vllg/chargebacks/chb_n9z0tp",
                "type": "application/hal+json"
            },
            "payment": {
                "href": "https://api.mollie.com/v2/payments/tr_WDqYK6vllg",
                "type": "application/hal+json"
            },
            "documentation": {
                "href": "https://docs.mollie.com/reference/v2/chargebacks-api/get-chargeback",
                "type": "text/html"
            }
        }
    }`);
    const chargeback = new Chargeback(chargebackProps);

    expect(chargeback.toPlainObject()).toMatchSnapshot();
  });
});
