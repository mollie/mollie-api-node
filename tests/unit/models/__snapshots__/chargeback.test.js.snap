// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chargeback model should instantiate with default values 1`] = `
Object {
  "_links": Object {
    "payment": null,
    "settlement": null,
  },
  "amount": null,
  "createdAt": null,
  "id": null,
  "paymentId": null,
  "resource": "chargeback",
  "reversedAt": null,
  "settlementAmount": null,
}
`;

exports[`chargeback model should instantiate with given values 1`] = `
Object {
  "_links": Object {
    "documentation": Object {
      "href": "https://docs.mollie.com/reference/v2/chargebacks-api/get-chargeback",
      "type": "text/html",
    },
    "payment": Object {
      "href": "https://api.mollie.com/v2/payments/tr_WDqYK6vllg",
      "type": "application/hal+json",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/payments/tr_WDqYK6vllg/chargebacks/chb_n9z0tp",
      "type": "application/hal+json",
    },
  },
  "amount": Object {
    "currency": "USD",
    "value": "43.38",
  },
  "createdAt": "2018-03-14T17:00:52.0Z",
  "id": "chb_n9z0tp",
  "paymentId": "tr_WDqYK6vllg",
  "resource": "chargeback",
  "reversedAt": null,
  "settlementAmount": Object {
    "currency": "EUR",
    "value": "-35.07",
  },
}
`;
